File: Python/2410. Maximum Matching of Players With Trainers.py
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        res = 0

        players.sort()
        trainers.sort()

        i, j = 0, 0
        lp, lt = len(players), len(trainers)

        while i < lp and j < lt:
            if players[i] <= trainers[j]: i += 1; res += 1
            j += 1

        return res

File: C/2410. Maximum Matching of Players With Trainers.c
int comp(const void* a, const void* b) {
    int t1 = *(int*)a, t2 = *(int*)b;

    if (t1 > t2) return 1;
    else if (t1 < t2) return -1;
    return 0;
}

int matchPlayersAndTrainers(int* players, int playersSize, int* trainers, int trainersSize){
    int res = 0;
    int i = 0, j = 0;

    qsort(players, playersSize, sizeof(int), comp);
    qsort(trainers, trainersSize, sizeof(int), comp);

    while (i < playersSize && j < trainersSize) {
        if (*(players + i) <= *(trainers + j)) {
            i++;
            res++;
        }

        j++;
    }

    return res;
}
